crates/module-system/sov-modules-api/src/transaction/mod.rs

```diff
@@ -66,6 +71,127 @@ pub struct Version0<Call, S: Spec> {
     pub details: TxDetails<S>,
 }
 
+#[derive(
+    derive_more::Debug,
+    Clone,
+    borsh::BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    borsh::BorshSerialize,
+    UniversalWallet,
+    PartialEq,
+    Eq,
+)]
+/// A signature and public key pair.
+pub struct PubKeyAndSignature<S: Spec> {
+    /// The signature.
+    pub signature: <S::CryptoSpec as CryptoSpec>::Signature,
+    /// The public key
+    pub pub_key: <S::CryptoSpec as CryptoSpec>::PublicKey,
+}
+
+impl<S: Spec> PubKeyAndSignature<S> {
+    /// Returns a reference to the public key.
+    pub fn key(&self) -> &<S::CryptoSpec as CryptoSpec>::PublicKey {
+        &self.pub_key
+    }
+}
+
+#[derive(
+    derive_more::Debug,
+    Clone,
+    borsh::BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    borsh::BorshSerialize,
+    UniversalWallet,
+)]
+#[serde(bound = "Call: serde::Serialize + serde::de::DeserializeOwned")]
+/// A V1 (multisig) transaction. The number of signers is capped at 10.
+///
+/// The credential ID for a multisig is hash(borsh(min_signers) || borsh(sort(pub_keys)))
+pub struct Version1<Call, S: Spec> {
+    /// The signatures of the transaction.
+    pub signatures: SafeVec<PubKeyAndSignature<S>, MAX_SIGNERS>,
+    /// The credential IDs that are part of the multisig but not used to sign the transaction.
+    // This is used to compute the credential ID statelessly
+    pub unused_pub_keys: SafeVec<<S::CryptoSpec as CryptoSpec>::PublicKey, MAX_SIGNERS>,
+    /// The minimum number of signers required to sign the transaction. In a 3/5 multisig, this would be 3.
+    /// Note that...
+    ///  - The transaction will be valid if at least `min_signers` signers have signed the transaction, but more signers are allowed. (This is useful for record keeping in case of, say, a unanimous decision)
+    ///  - If an invalid signature is detected, the transaction will be invalid. This allows us to use batch verification.
+    // This is used to compute the credential ID statelessly.
+    pub min_signers: u8,
+    /// The runtime call of the transaction.
+    #[sov_wallet(
+        bound = "Call: sov_rollup_interface::sov_universal_wallet::schema::UniversalWallet"
+    )]
+    pub runtime_call: Call,
+    /// Uniqueness identifier of this transaction. see [`UniquenessData`] for more details.
+    pub uniqueness: UniquenessData,
+    /// The transaction metadata. Contains gas parameters and the chain ID.
+    pub details: TxDetails<S>,
```

Comment:
If all supported transaction versions share the same `details, uniqueness, and runtime_call` fields, we can move these fields to the `Transaction` struct and reuse them across `V0` and `V1`.